docs/proposals/PROPOSAL_H.md

```diff
@@ -0,0 +1,352 @@
+
+    - **`compatibility feature field`**
+
+      This REQUIRED property describes a feature compatibility requirement on the target host, the field is named with OCI feature label.
+
+      The value of compatibility field in the compatibility artifact SHOULD be described in contiguous or discontinuous range if the value discovered on nodes can be different.
+
+      It's the tool's responsibility to interpret the value of OCI feature label, and matching criteria with the value discovered from the node via NFD tool or other tool. NFD means the Kubernetes node feature discovering SIG.
+
+      Inevitably, some compatibility fields MAY be duplicated among compatibility sets.
+
+      The value type of this field is STRING.
+
+    - **`compatibility score field`**
+
+      This OPTIONAL property describes the score of the compatibility set.
+
+      There is only one compatibility score field allowed in one compatibility set.
+
+      The highest score of the compatibility set means this compatibility set is the most optimal one.
+
+      If this field is not present, the score of this compatibility set is 0.
+
+      If compatibility sets have the same score, that means any of these compatibility sets pass the compatibility check is successful in matching the compatibility requirements.
+
+      The compatibility-score is used for selecting optimal image or optimal node in descending order for those compatibility sets pass the compatibility check.
+
+      The value type of this field is INT.
+
+    - **`annotation field`**
+
+      This OPTIONAL property is a field to describe human readable information regarding to the compatibility set.
+
+      The value type of this field is STRING.
+
+### Artifact example of use
+
+All fields in this section are only for examples.
+
+#### Example - image is compatible with different hardware combinations
+
+```json
+{
+  "compatibilities": [
+    {
+      "oci.cpu.vendor": "GenuineIntel",
+      "oci.cpu.features": "AVX512FP16,",
+      "oci.kernel.selinux": "false",
+      "oci.kernel.configurations": "PREEMPT,",
+      "oci.kernel.versions": "[4.19, 5.16)",
+      "oci.os.glibc": "[2.31, 2.37]",
+      "oci.pci.devices": "15B3.020D,",
+    },
+    {
+      "oci.cpu.vendor": "AuthenticAMD",
+      "oci.cpu.features": "FPHP,",
+      "oci.kernel.selinux": "false"
+      "oci.kernel.configurations": "PREEMPT,",
+      "oci.kernel.versions": "[4.19, 5.16)",
+      "oci.os.glibc": "[2.31, 2.37]",
+      "oci.pci.devices": "1002.67ff",
+      "annotation": "only work with AMD CPU and AMD GPU"
+    }
+  ]
+}
+```
+
+#### Example - image is compatible with optimal hardware
+
+```json
+{
+  "compatibilities": [
+    {
+      "oci.cpu.vendor": "GenuineIntel",
+      "oci.cpu.features": "AMXFP16,",
+      "oci.kernel.selinux": "false"
+      "oci.kernel.configurations": "PREEMPT,",
+      "oci.kernel.versions": "[4.19, 5.16)",
+      "oci.os.glibc": "[2.31, 2.37]",
```

Comment:
does this mean that the host system must be glibc based? how does one declare "glibc, any version?" 

Also, this does not handle the use case of long term / enterprise distributions that maybe backport functionality from a newer glibc release into an older glibc release that is installed on the system.